apollo-router/src/spec/query/tests.rs

```diff
@@ -1182,6 +1187,124 @@ fn reformat_response_array_of_id_duplicate() {
         .test();
 }
 
+#[test]
+// If this test fails, this means you got greedy about allocations,
+// beware of aliases!
+fn reformat_response_expected_int_got_string() {
+    FormatTest::builder()
+        .schema(
+            "type Query {
+                get: Thing
+            }
+            type Thing {
+                i: Int
+                s: String
+                f: Float
+                b: Boolean
+                e: E
+                u: U
+                id: ID
+                l: [Int]
+            }
+            
+            enum E {
+              A
+              B
+            }
+            union U = ObjA | ObjB
+            type ObjA {
+                a: String
+            }
+            type ObjB {
+                a: String
+            }
+            ",
+        )
+        .query(
+            r#"{
+            get {
+                i
+                s
+                f
+                ... on Thing {
+                  b
+                  e
+                  u {
+                    ... on ObjA {
+                      a
+                    }
+                  }
+                }
+                l
+            }
+        }"#,
+        )
+        .response(json! {{
+            "get": {
+                "i": "hello",
+                "s": 1.0,
+                "f": [1],
+                "b": 0,
+                "e": "X",
+                "u": 1,
+                // "id": {
+                //     "test": "test"
+                // }
```

Comment:
I assume this is commented out because it's not working? We are checking for the wrong type name in validation.
https://github.com/apollographql/router/blob/6881f9baebf17175cfcad4063d5a08c0a678e962/apollo-router/src/spec/query.rs#L589

It should be `"ID"`, not `"Id"`.